nck/readers/mytarget_reader.py

```diff
@@ -0,0 +1,408 @@
+from nck.helpers.mytarget_helper import REQUEST_CONFIG
+
+
+@click.command(name="read_mytarget")
+@click.option("--mytarget-client-id", required=True)
+@click.option("--mytarget-client-secret", required=True)
+@click.option("--mytarget-mail", required=True)
+@click.option("--mytarget-agency", required=True)
+@click.option("--mytarget-refresh-token", required=True)
+@click.option("--mytarget-start-date", type=click.DateTime(), required=True)
+@click.option("--mytarget-end-date", type=click.DateTime(), required=True)
+@processor("mytarget-client-id", "mytarget-client-secret")
+def mytarget(**kwargs):
+    return MyTargetReader(**extract_args("mytarget_", kwargs))
+
+
+class MyTargetReader(Reader):
+
+    def __init__(
+        self,
+        client_id,
+        client_secret,
+        mail,
+        agency,
+        refresh_token,
+        **kwargs
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.mail = mail
+        self.agency = agency
+        self.agency_client_token = {'refresh_token': refresh_token}
+        self.start_date = kwargs.get('start_date')
+        self.end_date = kwargs.get('end_date')
+        self.date_format = kwargs.get('date_format')
+        self.day_range = kwargs.get('day_range')
+
+    def read(self):
+        if self.__check_date_input_validity():
+            self.__retrieve_and_set_token()
+
+            rsp_daily_stat, names_dict, ids_dict, rsp_banner_names = self.__retrieve_all_data()
+
+            complete_daily_content = self.map_campaign_name_to_daily_stat(
+                rsp_daily_stat,
+                names_dict,
+                ids_dict,
+                rsp_banner_names
+            )
+
+            yield JSONStream(
+                "results_", self.split_content_by_date(complete_daily_content)
+            )
+
+    def __check_date_input_validity(self) -> bool:
+        """The goal of this function is to check the validity of the date input parameters before retrieving the data.
+        """
+        return self.__check_both_start_end_valid(self.start_date, self.end_date) and \
+            self.__check_validity_date(self.start_date) & self.__check_validity_date(self.end_date) and \
+            self.__check_end_posterior_to_start(self.start_date, self.end_date) and \
+            self.__check_date_not_in_future(self.end_date)
+
+    def __is_none(self, date: datetime) -> bool:
+        return date is None
+
+    def __check_validity_date(self, date: datetime) -> bool:
+        try:
+            datetime(date.year, date.month, date.day)
+            return True
+        except ValueError as e:
+            raise ValueError(f'The date is not valid : {e}')
+
+    def __check_date_not_in_future(self, end_date: datetime) -> bool:
+        if end_date <= datetime.now():
+            return True
+        else:
+            raise ValueError(f'The end date {end_date} is posterior to current date {datetime.now()}')
+
+    def __check_both_start_end_valid(
```

Comment:
Same here, this check is unnecessary because there is the `required=True` option.